File: C#/T1/Project/PetSimulator/Player.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace PetSimulator
{
    class Player
    {
        public string playerName { get; set; }

        public Player()
        {
            playerName = "DEFAULT NAME";
        }

        public void feedingPet(Pet p, float amount)
        {
            p.Eat(amount);
        }

        public void goForWalk(Pet p, float distance)
        {
            p.Walk(distance);
        }

        public void goForSleep(Pet p, float amount)
        {
            p.Sleep(amount);
        }

        public void goForPlay(Pet p, float amount)
        {
            p.Playing(amount);
        }

        public void goForMovie(Pet p, float amount)
        {
            p.WatchMovie(amount);
        }

        public void goForToilet(Pet p)
        {
            p.Toilet();
        }
    }
}

File: C++/T2/PetSimulator/PetSimulator/Player.cpp
#include "Player.h"

Player::Player() :
	playerName("DEFAULT NAME")
{}

Player::~Player()
{}

void Player::setName(std::string name)
{
	playerName = name;
}

std::string Player::getName()
{
	return playerName;
}

void Player::feedingPet(Pet * p, float amount) // In Gram
{
	p->Eat(amount);
}

void Player::goForWalk(Pet * p, float distance) // In Meter
{
	p->Walk(distance);
}

void Player::goForSleep(Pet * p, float amount) // In Hour
{
	p->Sleep(amount);
}

void Player::goForPlay(Pet * p, float amount) // In Hour
{
	p->Playing(amount);
}

void Player::goForMovie(Pet * p, float amount) // In Hour
{
	p->WatchMovie(amount);
}

void Player::goForToilet(Pet * p)
{
	p->Toilet();
}
